Mage.Sets/src/mage/cards/c/CecilDarkKnight.java

```diff
@@ -0,0 +1,107 @@
+package mage.cards.c;
+
+import java.util.Optional;
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.game.Controllable;
+import mage.game.Game;
+import mage.players.Player;
+import mage.abilities.Ability;
+import mage.abilities.common.DealsDamageSourceTriggeredAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.dynamicvalue.common.SavedDamageValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.LoseLifeSourceControllerEffect;
+import mage.abilities.effects.common.TransformSourceEffect;
+import mage.abilities.effects.common.UntapSourceEffect;
+import mage.abilities.keyword.DeathtouchAbility;
+import mage.abilities.keyword.TransformAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+
+/**
+ * @author balazskristof
+ */
+public final class CecilDarkKnight extends CardImpl {
+
+    public CecilDarkKnight(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{B}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.KNIGHT);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(3);
+
+        this.secondSideCardClazz = mage.cards.c.CecilRedeemedPaladin.class;
+
+        // Deathtouch
+        this.addAbility(DeathtouchAbility.getInstance());
+
+        // Darkness -- Whenever Cecil deals damage, you lose that much life. Then if your life total is less than or equal to half your starting life total, untap Cecil and transform it.
+        this.addAbility(new TransformAbility());
+        Ability ability = new DealsDamageSourceTriggeredAbility(new LoseLifeSourceControllerEffect(SavedDamageValue.MUCH), false).withFlavorWord("Darkness");
+        ability.addEffect(new ConditionalOneShotEffect(
+            new CecilDarkKnightEffect(),
```

Comment:
you can use `UntapSourceEffect` and `ConditionalOneShotEffect::addEffect(new UntapSourceEffect())` instead of a custom effect